pkg/kv/kvpb/batch.go

```diff
@@ -714,7 +714,7 @@ func (br *BatchResponse) Combine(
 	for i := range otherBatch.Responses {
 		pos := positions[i]
 		if br.Responses[pos] == (ResponseUnion{}) {
-			br.Responses[pos] = otherBatch.Responses[i]
+			br.Responses[pos].MustSetInner(otherBatch.Responses[i].GetInner().ShallowCopy())
```

Comment:
I'm not against merging this, btw, if this doesn't cause a regression on `BenchmarkSysbench/SQL/3node/oltp_read_write` and you leave a paper trail linking to our discussion and the original data race.